File: solutions/fsharp/lucians-luscious-lasagna/1/LuciansLusciousLasagna.fs
module LuciansLusciousLasagna

let expectedMinutesInOven = 40

let remainingMinutesInOven x = expectedMinutesInOven - x

let preparationTimeInMinutes x = 2 * x

let elapsedTimeInMinutes layer minsInOven = preparationTimeInMinutes layer + minsInOven

File: solutions/csharp/lucians-luscious-lasagna/2/LuciansLusciousLasagna.cs
class Lasagna
{
    public int ExpectedMinutesInOven() => 40;

    public int RemainingMinutesInOven(int actual_minutes_in_oven){
        return ExpectedMinutesInOven() - actual_minutes_in_oven;
    }

    public int PreparationTimeInMinutes(int layers){
        return 2*layers;
    }

    public int ElapsedTimeInMinutes(int layers, int actual_minutes_in_oven){
        return PreparationTimeInMinutes(layers)+actual_minutes_in_oven;
    }
}
